FIX Allow editable form fields to have nullable titles rather than fallback to Name

robbieaverill opened the pull request:
Fixes #753

dhensby commented:
getting some failures due to bad docblocks in tests:

>Trying to @cover or @use not existing method "EditableOption::getValue".

robbieaverill commented:
Updated (removed the incorrect \@covers tag) and also switched a functional test assertion to handle the fact that SS 4.2 now adds `?stage=Stage` for draft content to URLs.
